src/seedu/addressbook/data/person/UniquePersonList.java

```diff
@@ -81,6 +76,36 @@ public UniquePersonList(UniquePersonList source) {
         return Collections.unmodifiableList(internalList);
     }
 
+    /**
+     * Returns a java List view with elements cast as Person
+     * For use with other methods/libraries.
+     * Any changes to the internal list/elements are immediately visible in the returned list.
+     */
+    public List<Person> sortedListView() {
+        List<Person> newInternalList = new ArrayList<>();
+        for (Person p : internalList) {
+            newInternalList.add(p);
+        }
+        Collections.sort(newInternalList, new Comparator<Person>() {
+            @Override
+            public int compare(Person person1, Person person2) {
+                String name1 = person1.getName().fullName;
+                String name2 = person2.getName().fullName;
+                return name1.compareToIgnoreCase(name2);
+            }
+        });
+        return newInternalList;
+    }
+
+    /**
+     * Returns a java List view with elements cast as Person.
+     * For use with AdressBookTest.java only.
+     * Any changes to the internal list/elements are immediately visible in the returned list.
+     */
+    public List<Person> listView() {
+        return internalList;
+    }
```

Comment:
What is the difference between this method and `immutableListView` above? Is it really necessary for you to have access to `Person` instead of `ReadOnlyPerson`?

Reply:
No, I should have left it as ReadOnlyPerson.